services/workflows-service/src/workflow/hook-callback-handler.service.ts

```diff
@@ -111,18 +118,41 @@ export class HookCallbackHandlerService {
 
     if (!business) throw new BadRequestException('Business not found.');
 
-    await this.businessReportService.create({
-      data: {
-        type: reportType as BusinessReportType,
-        report: {
-          reportFileId: pdfReportBallerineFileId,
-          data: reportData as InputJsonValue,
+    const existantBusinessReport = await this.businessReportService.findFirst(
+      {
+        where: {
+          businessId: business.id,
           reportId: reportId as string,
         },
-        businessId: business.id,
-        projectId: currentProjectId,
       },
-    });
+      [currentProjectId],
+    );
+
+    await this.businessReportService.upsert(
+      {
+        create: {
+          type: reportType as BusinessReportType,
+          report: {
+            reportFileId: pdfReportBallerineFileId,
+            data: reportData as InputJsonValue,
+          },
+          reportId: reportId as string,
+          businessId: business.id,
+          projectId: currentProjectId,
+        },
+        update: {
+          type: reportType as BusinessReportType,
+          report: {
+            reportFileId: pdfReportBallerineFileId,
+            data: reportData as InputJsonValue,
+          },
+        },
+        where: {
+          id: existantBusinessReport?.id,
+        },
```

Comment:
add project in here as well...
in case 'id' is undefined you will face with unexpected behavior

Reply:
project id is being handled by scope.
therefore you don't need ( a column below )